rust/agents/scraper/src/agent.rs

```diff
@@ -152,7 +152,7 @@ impl BaseAgent for Scraper {
     }
 
     #[allow(clippy::async_yields_async)]
-    async fn run(&self) -> Instrumented<JoinHandle<eyre::Result<()>>> {
+    async fn run(self) -> Instrumented<JoinHandle<eyre::Result<()>>> {
```

Comment:
Tbh, I have no idea why I didn't already have it take the `self` receiver. This makes things slightly easier since we can avoid a mutex in the validator.